pydantic_ai_slim/pydantic_ai/models/bedrock.py

```diff
@@ -910,7 +1017,12 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
             self.provider_response_id = self._provider_response_id
 
         chunk: ConverseStreamOutputTypeDef
-        tool_id: str | None = None
+        tool_ids: dict[int, str] = {}
+
+        # Bedrock has deltas for built-in tool returns, which aren't supported by parts manager.
+        # We accumulate the deltas here and yield the complete return part once the content block ends
+        builtin_tool_returns: dict[int, BuiltinToolReturnPart] = {}
```

Comment:
Parts manager doesn't seem to support deltas for BuiltinToolReturnPart(s), and from what I can tell from the other model tests, they only emit a single `PartStartEvent` for a BuiltinToolReturnPart.  Other models didn't seem to emit any `PartDeltaEvent`(s), or `PartEndEvent`(s).

On the other hand, Bedrock/Nova2 seems to have multiple blocks for a  BuiltinToolReturnPart.  So here I'm just accumulating them and then fire one event in the `contentBlockStop` case (via `yield self._parts_manager.handle_part`)